### DISCUSSION TITLE
Need help in setting up in Synology Docker

### DISCUSSION THREAD

**Discussion by Whenesday** [Difegue/LANraragi#734] [Q&A] [answered]
Hello, I'm having difficulties in accessing Lanraragi via serverIP:3000 after setup. This is what I've done so far:

1: Searched for Lanraragi image in Docker Registry list (https://registry.hub.docker.com/r/difegue/lanraragi/), and click to install
2: Selected "bridge" as default container
![image](https://user-images.githubusercontent.com/6242699/209895594-7d72f5ab-815c-4c8b-8f1b-2c900cf12e76.png)
3: Went for default in General and Advanced Environment and
![image](https://user-images.githubusercontent.com/6242699/209895705-2ee3113a-b059-4a48-bdf5-2f3ca8ae4b6f.png)

![image](https://user-images.githubusercontent.com/6242699/209895634-34448969-88de-4062-b2c8-e5ffa00730bf.png)
4: Default in port Settings, 3000 container port
![image](https://user-images.githubusercontent.com/6242699/209895733-7fc13838-d924-4a9d-8394-d5e7471d5345.png)
5: Default in Volume settings, none, and proceeded to create and run docker container
![image](https://user-images.githubusercontent.com/6242699/209895768-f2351cb6-536f-4c04-8001-17e5028c508e.png)
6: Logs seems to show that it's running correctly![image](https://user-images.githubusercontent.com/6242699/209895841-930f6fab-895f-455b-bf5f-83abfc44f733.png)
7: But when I tried to access my NASLocalIP:3000, it's not accessing Lanraragi page. 

Am I missing something?

**Comment by dar4545** [comment, marked as answer]
Docker version requires few more manual setting when installation. 

**from step 3:**
![209895705-2ee3113a-b059-4a48-bdf5-2f3ca8ae4b6f](https://user-images.githubusercontent.com/23472689/210173083-a7114f9a-e6a0-4fda-966a-1a0ec38ac6ff.png)

![209895634-34448969-88de-4062-b2c8-e5ffa00730bf](https://user-images.githubusercontent.com/23472689/210173100-b30018cc-4dfb-468e-9395-fa368646fb53.png)

For finding your uid and gid, you can search from Google or follow this guide.
https://mariushosting.com/synology-find-uid-userid-and-gid-groupid-in-5-seconds/

**from step 4:**
![209895733-7fc13838-d924-4a9d-8394-d5e7471d5345](https://user-images.githubusercontent.com/23472689/210173203-3c2d9b39-7248-4390-ae64-e2d6105a85f6.png)

**from step 5:**
You need to create or assign two directories on your NAS for storing LANraragi database and your book contents.
Example of mine: (the Mount Path are fixed values)
![000](https://user-images.githubusercontent.com/23472689/210173354-36bc516a-fc94-46e3-8ab7-11ab99f24ce3.jpg)

Also, the permission of the directories may need to allow read & write for everyone.
![000](https://user-images.githubusercontent.com/23472689/210173392-13ae1f87-c382-4540-95ac-0aa14c500c44.jpg)


Hope my set up could help you.

**Reply by doublewelp** [reply]
You shouldn't need the high privilege flag - the default permissions provided by the container should be enough. I'm running Lanraragi on a DS920+ with the Synology provided Docker UI and the only snag was making sure the docker user had read/write permissions on the shared folder:
![imagem](https://user-images.githubusercontent.com/106932725/210179683-19c1a528-3787-45e6-b92e-f1d1826ebfc3.png)
You can change these in the Control Panel, Shared Folder option.

**Reply by Whenesday** [reply]
Thanks for the help! 
It seems that adding the local port 3000 is good enough to make it accessible :)
